tests/rptest/services/kcat_consumer.py

```diff
@@ -0,0 +1,274 @@
+                f"sasl.mechanism={cfg['sasl_mechanism']}", "-X",
+                f"sasl.username={cfg['sasl_plain_username']}", "-X",
+                f"sasl.password={cfg['sasl_plain_password']}"
+            ]
+            if cfg['sasl_mechanism'] == "GSSAPI":
+                cmd += [
+                    "-X", "sasl.kerberos.service.name=redpanda",
+                    '-Xsasl.kerberos.kinit.cmd=kinit client -t /var/lib/redpanda/client.keytab'
+                ]
+        self._cmd += trailopts
+
+    class SSHCapturedPipes:
+        def __init__(self, stdin: paramiko.channel.ChannelStdinFile,
+                     stdout: paramiko.channel.ChannelFile,
+                     stderr: paramiko.channel.ChannelStderrFile, cmd: str,
+                     logger) -> None:
+            self.stdin = stdin
+            self.stdout = stdout
+            self.stderr = stderr
+            self._cmd = cmd
+            self.logger = logger
+
+        def __enter__(self) -> 'KcatConsumer.SSHCapturedPipes':
+            return self
+
+        def __exit__(self, exc_type, exc_value, traceback) -> None:
+            try:
+                exit_status = self.stdout.channel.recv_exit_status()
+                if exit_status != 0:
+                    if self.logger is not None:
+                        #self._log(logging.DEBUG,
+                        self._redpanda.logger.debug(
+                            "Running ssh command '%s' exited with status %d and message: %s"
+                            % (self._cmd, exit_status, self.stderr.read()))
+                    else:
+                        raise RemoteCommandError(self, self._cmd, exit_status,
+                                                 self.stderr.read())
+            finally:
+                self.stdin.close()
+                self.stdout.close()
+                self.stderr.close()
+
+    def ssh_capture_with_pipes(self,
+                               node,
+                               cmd,
+                               allow_fail=False,
+                               timeout_sec=None):
+        #self._log(logging.DEBUG, "Running ssh command: %s" % cmd)
+        self._redpanda.logger.debug("Running ssh command: %s" % cmd)
+
+        client = node.account.ssh_client
+        chan = client.get_transport().open_session(timeout=timeout_sec)
+
+        chan.settimeout(timeout_sec)
+        chan.exec_command(cmd)
+        chan.set_combine_stderr(False)
+
+        stdin = chan.makefile('wb', -1)  # set bufsize to -1
+        stdout = chan.makefile('r', -1)
+        stderr = chan.makefile_stderr('r', -1)
+
+        return KcatConsumer.SSHCapturedPipes(
+            stdin, stdout, stderr, cmd,
+            self._redpanda.logger if allow_fail else None)
+
+        def output_generator():
+            for line in iter(stdout.readline, ''):
+                yield line
+
+        return SSHOutputIter(output_generator, stdout)
```

Comment:
This indenting is weirding me out for some reason.  It appears to me that you're returning out of `ssh_capture_with_pipes` and then declaring `output_generator()` and then returning `SSHOutputIter`.  Or are those last two part of something else that I'm missing?